eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/PageableWrapperProvider.cs

```diff
@@ -0,0 +1,197 @@
+                Name: "AsPages",
+                Description: $"Converts the pages from {sourceTypeName} to {pageUType.Name}.",
+                Modifiers: modifiers,
+                ReturnType: returnType,
+                ReturnDescription: null,
+                Parameters: new[] { continuationTokenParam, pageSizeHintParam });
+
+            // Build the method body using foreach and yield return pattern
+            var pageType = new CSharpType(typeof(Page<>), _tType);
+            var foreachStatement = new ForEachStatement(
+                pageType,
+                "page",
+                _sourceField.Invoke("AsPages", [continuationTokenParam, pageSizeHintParam]),
+                isAsync: _isAsync,
+                out var pageVar);
+
+            // Create a list to collect converted values instead of using LINQ Select
+            var listType = new CSharpType(typeof(List<>), _uType);
+            var listDeclaration = Declare("convertedItems", listType, New.Instance(listType), out var listVar);
+
+            // Create inner foreach to convert each item
+            var innerForeachStatement = new ForEachStatement(
+                _tType,
+                "item",
+                pageVar.Property("Values"),
+                isAsync: false,
+                out var itemVar);
+
+            innerForeachStatement.Add(listVar.Invoke("Add", [_converterField.Invoke("Invoke", [itemVar])]).Terminate());
+
+            foreachStatement.Add(listDeclaration);
+            foreachStatement.Add(innerForeachStatement);
+            foreachStatement.Add(
+                new YieldReturnStatement(
+                    Static(pageUType).Invoke(
+                        "FromValues",
+                        [
+                            listVar,
+                            pageVar.Property("ContinuationToken"),
+                            pageVar.Invoke("GetRawResponse", [])
+                        ])));
+
+            var bodyStatements = new MethodBodyStatement[] { foreachStatement };
+
+            return new MethodProvider(signature, bodyStatements, this);
+        }
+
+        protected override ConstructorProvider[] BuildConstructors()
+        {
+            var sourceParameterType = _isAsync ? _asyncPageableT : _pageableT;
+            var sourceParameterDescription = _isAsync
+                ? (FormattableString)$"The source async pageable value of type AsyncPageable<{_tType.Name}>."
+                : (FormattableString)$"The source pageable value of type Pageable<{_tType.Name}>.";
+
+            var sourceParameter = new ParameterProvider("source", sourceParameterDescription, sourceParameterType);
+            var converterParameter = new ParameterProvider("converter", $"The converter function from {_tType.Name} to {_uType.Name}.", new CSharpType(typeof(Func<,>), _tType, _uType));
+
+            var signature = new ConstructorSignature(
+                Type,
+                null,
```

Comment:
is this the place throwing error if we set xml document? add a comment for it, ideally we should file an issue for MTG